pkg/servicescm/servicescm.go

```diff
@@ -119,6 +127,106 @@ func Generate(name, namespace string) (*core.ConfigMap, error) {
 	return servicesConfigMap, nil
 }
 
+// Parse converts ConfigMap data into the objects representing a Windows services ConfigMap schema
+// Returns error if the given data is invalid in structure or contents.
+func Parse(data map[string]string) (*data, error) {
```

Comment:
Input and output are both called `data`. I suggest using `dataFromCM` as the function parameter.

I also suggest squashing the two validation commits, given they are actually one logical unit i.e. introduce `Parse()` and use it in the same commit.